File: kotlin/hamming/src/main/kotlin/Hamming.kt
package Hamming

fun compute(dna1: String, dna2: String): Int {
    require(dna1.length == dna2.length) {
        "left and right strands must be of equal length."
    }


    var count = 0
    for ((index, nucleotide1) in dna1.withIndex()) {
        val nucleotide2 = dna2[index]
        if (nucleotide1 != nucleotide2) {
            count++
        }
    }
    return count
}

File: javascript/hamming/hamming.js
/**
 * @prettier
 */

export const compute = (left, right) => {
  if (left.length !== right.length) {
    if (left.length === 0) throw new Error("left strand must not be empty");
    if (right.length === 0) throw new Error("right strand must not be empty");

    throw new Error("left and right strands must be of equal length");
  }

  return left.split("").reduce((count, leftCharacter, index) => {
    return leftCharacter != right[index] ? count + 1 : count;
  }, 0);
};
